### ISSUE TITLE
Issue with Chip-seq pipeline: jobid: 27: one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!

### ISSUE THREAD

**Issue by Jerolen** [vanheeringen-lab/seq2science#939]
**Question**
Hi There, I am trying to test out the chip-set pipeline using a single sample of paired end data but the workflow keeps failing at a specific point with the following error:

Activating conda environment: ../opt/miniconda3/envs/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/6824310427e127c1e8fdddc2f1df8a12
/bin/bash: line 1: shuf: command not found
[Wed Dec  7 15:20:57 2022]
Error in rule random_subset_peaks:
    jobid: 27
    output: /Users/jerolen/my_project2/results/qc/computeMatrix_peak/GRCh38-macs2_N20000.bed
    conda-env: /Users/jerolen/opt/miniconda3/envs/seq2science/lib/python3.8/site-packages/seq2science/.snakemake/6824310427e127c1e8fdddc2f1df8a12
    shell:
        
        shuf -n 20000 /Users/jerolen/my_project2/results/macs2/GRCh38_combinedsummits.bed | bedtools slop -i stdin -g /Users/jerolen/my_project2/genomes/GRCh38/GRCh38.fa.sizes -b 1000 > /Users/jerolen/my_project2/results/qc/computeMatrix_peak/GRCh38-macs2_N20000.bed
        
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

**What have I tried**
I have tried changing the alignment tool from bwa to hisat2 with no luck. Any advice would be greatly appreciated as I am completely out of depth on the coding side of things.  config.yaml, samples.tsv and export of terminal attached. Thank you. 

[config-yaml.txt](https://github.com/vanheeringen-lab/seq2science/files/10176584/config-yaml.txt)
[samples.txt](https://github.com/vanheeringen-lab/seq2science/files/10176614/samples.txt)
[Terminal Saved Output-Jerolen-Seq2SCience-chipseq.txt](https://github.com/vanheeringen-lab/seq2science/files/10176624/Terminal.Saved.Output-Jerolen-Seq2SCience-chipseq.txt)

**Comment by Maarten-vd-Sande** [comment]
Thanks for the issue! I think there's actually a problem on "our" side here. Seems like we haven't defined the environment properly and we are making use of the `shuf` command which isn't by default on all systems (e.g. on a mac)...

Let me take a look!

For now you can turn off the deeptools figures for the QC report as this is purely used for visualization only. In the config.yaml add:

`deeptools_qc: false`

**Comment by Maarten-vd-Sande** [comment]
I don't have a mac, so you will have to check for me if this solution fixes it :smile: 

This file should already exists:
```
/opt/miniconda3/envs/seq2science/lib/python3.8/site-packages/seq2science/envs/bedtools.yaml
```

Can you change it so that it has coreutils added:

```
name: bedtools
channels:
  - conda-forge
  - bioconda
  - defaults
dependencies:
  - bioconda::bedtools=2.29.2
  - conda-forge::coreutils=9.1
  - conda-forge::conda-ecosystem-user-package-isolation=1.0
 ```

And then try seq2science again?

**Comment by Maarten-vd-Sande** [comment]
Have you had the time to test if this works?

**Comment by Jerolen** [comment]
Hi Maarten

Thanks for the follow up. I tried running it a a few more times without success. I'm not sure if my system runs out of RAM at the end.

I was running it locally, with 4 cores and 16GB RAM but I noticed that before the crashes it would say available memory at 3GB.

Any advice would be greatly appreciated.

Kind Regards

Jerolen

Jerolen Naidoo PhD
Senior Researcher
Bioengineering & Integrated Genomics Group (BIGG)
CSIR NextGen Health
Building 20
Meiring Naude road
Brummeria
Pretoria
0002

________________________________

**Comment by Maarten-vd-Sande** [comment]
Not sure, 16GB isn't too much indeed for preprocessing.. Which step(s) is the one crashing? Still the `random_subset_peaks` rule? Most rules make a logfile, perhaps there is a clue about what goes wrong?

https://vanheeringen-lab.github.io/seq2science/content/faq.html#one-of-the-rules-failed-and-the-log-turns-red

**Comment by Maarten-vd-Sande** [comment]
Your issue(s) should be solved with the latest 0.9.7 release. Let me know if you encounter anything else!!!

**Comment by Jerolen** [comment]
Thanks Maarten

I just got back from the break so will try again and give you an update. All the best for 2023.

Kind Regards

Jerolen

Jerolen Naidoo, PhD

Senior Researcher
Bioengineering & Integrative Genomics Group (BIGG)
Next Generation Health Cluster
CSIR Building 20
Meiring Naude Road
Pretoria

Tel: 012 841 3940
Fax: 012 841 2615
website: https://csir.co.za/
________________________________
